packages/calcite-components/src/components/tree-item/tree-item.tsx

```diff
@@ -227,17 +231,27 @@ export class TreeItem implements ConditionalSlotComponent, InteractiveComponent
 
     const checkbox =
       this.selectionMode === "ancestors" ? (
-        <label class={CSS.checkboxLabel} key="checkbox-label">
-          <calcite-checkbox
-            checked={this.selected}
+        <div class={CSS.checkboxContainer}>
+          <div
+            aria-checked={toAriaBoolean(this.selected)}
+            aria-disabled={this.disabled}
+            aria-label={this.label}
             class={CSS.checkbox}
-            data-test-id="checkbox"
-            indeterminate={this.hasChildren && this.indeterminate}
-            scale={this.scale}
             tabIndex={-1}
-          />
-          {defaultSlotNode}
-        </label>
+          >
+            <calcite-icon
+              icon={
+                this.selected
+                  ? ICONS.checkSquareF
+                  : checkboxIsIndeterminate
+                    ? ICONS.minusSquareF
+                    : ICONS.square
+              }
+              scale={getIconScale(this.scale)}
+            />
```

Comment:
Add `aria-hidden` so AT doesn't pick up the icon:

```suggestion
            <calcite-icon
            aria-hidden="true"
              icon={
                this.selected
                  ? ICONS.checkSquareF
                  : checkboxIsIndeterminate
                    ? ICONS.minusSquareF
                    : ICONS.square
              }
              scale={getIconScale(this.scale)}
            />
```

Reply:
Refer to https://github.com/Esri/calcite-design-system/pull/9849#discussion_r1695931093, as this is not needed since its already honored on the `icon` component.